File: performance/test-project/src/native/java/performance/sample/nativeimpl/FixedClock.java
package performance.sample.nativeimpl;

import dev.capylang.NativeImplementation;
import performance.sample.nativeinterop.NativeClock;

@NativeImplementation(qualifier = "fixed")
public final class FixedClock implements NativeClock {
    @Override
    public long fixed_millis() {
        return 4242L;
    }
}

File: performance/test-project/src/native/py/performance/sample/nativeimpl/FixedClock.py
def NativeImplementation(_target=None, **_kwargs):
    def decorate(cls):
        return cls
    if _target is None:
        return decorate
    return decorate(_target)


@NativeImplementation(qualifier="fixed", interface_id="performance.sample.nativeinterop.NativeClock")
class FixedClock:
    def fixed_millis(self):
        return 4242
